packages/cli/src/commands/types/generate/actions.test.ts

```diff
@@ -815,3 +818,138 @@ describe('generateStoryblokTypes', () => {
     expect(savedContent).toContain('export interface StoryblokCustom');
   });
 });
+
+describe('separate files mode', () => {
+  it('returns an array of files when options.separateFiles = true (one per component)', async () => {
+    const spaceData: SpaceComponentsData = {
+      components: [
+        {
+          name: 'Hero Section X',
+          display_name: 'Hero',
+          created_at: '2023-01-01T00:00:00Z',
+          updated_at: '2023-01-01T00:00:00Z',
+          id: 1,
+          schema: {
+            title: { type: 'text', required: true },
+          },
+          internal_tags_list: [],
+          internal_tag_ids: [],
+        },
+        {
+          name: 'cta-button',
+          display_name: 'CTA Button',
+          created_at: '2023-01-01T00:00:00Z',
+          updated_at: '2023-01-01T00:00:00Z',
+          id: 2,
+          schema: {
+            href: { type: 'multilink', required: false, email_link_type: true, asset_link_type: true },
+          },
+          internal_tags_list: [],
+          internal_tag_ids: [],
+        },
+      ],
+      datasources: [],
+      groups: [],
+      presets: [],
+      internalTags: [],
+    };
+
+    const result = await generateTypes(spaceData, { separateFiles: true, strict: false });
+    expect(Array.isArray(result)).toBe(true);
+    const files = result as { name: string; content: string }[];
+    expect(files).toHaveLength(2);
+
+    expect(files[0].content).toContain('// This file was generated by the storyblok CLI.');
+    expect(files[0].content).toContain('// DO NOT MODIFY THIS FILE BY HAND.');
+    expect(files[0].name).toBe('Hero Section X')
+    expect(files.some(f => f.content.includes('export interface HeroSectionX'))).toBe(true);
```

Comment:
nitpick: why did you use `some` here instead of explicitly selecting a file with `files[0]`?

Reply:
Good catch. I was coding a bit late at night at the time of this commit 😁